packages/cli/yaml/validator/src/rules/no-response-property/no-response-property.ts

```diff
@@ -29,30 +29,50 @@ export const NoResponsePropertyRule: Rule = {
                     });
                     const responseBodyType = file.parseTypeReference(response);
                     const responseProperty = typeof response !== "string" ? response.property : undefined;
-                    if (
-                        responseProperty != null &&
-                        !typeReferenceHasProperty(responseBodyType, responseProperty, file, typeResolver)
-                    ) {
-                        return [
-                            {
-                                severity: "error",
-                                message: `Response does not have a property named ${responseProperty}.`,
-                            },
-                        ];
+                    if (responseProperty == null) {
+                        return [];
                     }
-                    return [];
+                    const result = typeReferenceHasProperty(responseBodyType, responseProperty, file, typeResolver);
+                    return resultToRuleViolations(result, responseProperty);
                 },
             },
         };
     },
 };
 
+enum Result {
+    ContainsProperty,
+    DoesNotContainProperty,
+    IsNotObject,
+}
+
+function resultToRuleViolations(result: Result, responseProperty: string): RuleViolation[] {
+    switch (result) {
+        case Result.ContainsProperty:
+            return [];
+        case Result.DoesNotContainProperty:
+            return [
+                {
+                    severity: "error",
+                    message: `Response does not have a property named ${responseProperty}.`,
+                },
+            ];
+        case Result.IsNotObject:
+            return [
+                {
+                    severity: "error",
+                    message: `Response does not have a property named ${responseProperty}.`,
```

Comment:
my guess is you want the error message to be different here, but not too opinionated. 

Something like: `Response must be an object in order to return a property as a response.` ?

Reply:
Gah, my mistake (again)! I actually meant to use the old error message we had here:

```
throw new Error("A custom response property is only supported for objects");
```

I like your error message better tho, so I'll update to that 👍